File: status/src/lib/domainUtils.js
/**
 * Format a name into domain-friendly format (lowercase with hyphens)
 * @param {string} name - Name to format
 * @returns {string} Formatted domain-friendly name
 */
export const formatDomainName = (name) => {
  return name.toLowerCase().replace(/\s+/g, '-');
};

/**
 * Format a full domain with latency.space suffix
 * @param {string} name - Name to format
 * @returns {string} Full domain name
 */
export const formatFullDomain = (name) => {
  return `${formatDomainName(name)}.latency.space`;
};

/**
 * Format a moon domain with its parent planet
 * @param {string} moonName - Moon name
 * @param {string} planetName - Parent planet name
 * @returns {string} Formatted moon domain
 */
export const formatMoonDomain = (moonName, planetName) => {
  return `${formatDomainName(moonName)}.${formatDomainName(planetName)}.latency.space`;
};

/**
 * Format a target domain with a celestial body
 * @param {string} targetDomain - Target domain name
 * @param {string} celestialName - Celestial body name
 * @returns {string} Formatted target domain
 */
export const formatTargetDomain = (targetDomain, celestialName) => {
  return `${targetDomain}.${formatDomainName(celestialName)}.latency.space`;
};

File: test-docker-build/domain_utils.go
package main

import (
	"fmt"
	"strings"
)

// FormatDomainName formats the name of a celestial body or spacecraft into a valid domain name
// It converts the name to lowercase and replaces spaces with hyphens
func FormatDomainName(name string) string {
	return strings.ToLower(strings.ReplaceAll(name, " ", "-"))
}

// FormatFullDomain formats a full domain with the latency.space suffix
func FormatFullDomain(name string) string {
	return fmt.Sprintf("%s.latency.space", FormatDomainName(name))
}

// FormatMoonDomain formats a moon domain with its parent planet
func FormatMoonDomain(moonName, planetName string) string {
	return fmt.Sprintf("%s.%s.latency.space", FormatDomainName(moonName), FormatDomainName(planetName))
}

// FormatTargetDomain formats a target domain with a celestial body
// Example: example.com.mars.latency.space
func FormatTargetDomain(targetDomain, celestialName string) string {
	return fmt.Sprintf("%s.%s.latency.space", targetDomain, FormatDomainName(celestialName))
}

// FormatMoonTargetDomain formats a target domain with a moon and its parent planet
// Example: example.com.europa.jupiter.latency.space
func FormatMoonTargetDomain(targetDomain, moonName, planetName string) string {
	return fmt.Sprintf("%s.%s.%s.latency.space", targetDomain, FormatDomainName(moonName), FormatDomainName(planetName))
}
